packages/nextjs/app/feed/richTextEditor/plugins/EditorContentPlugin.js

```diff
@@ -1,17 +1,20 @@
 import { useContext, useEffect } from "react";
 import { EditorContext } from "../../context";
+import { $generateHtmlFromNodes } from "@lexical/html";
 import { useLexicalComposerContext } from "@lexical/react/LexicalComposerContext";
 
 export default function EditorContentPlugin() {
   const [editor] = useLexicalComposerContext();
   const { setContent: updateContent } = useContext(EditorContext);
 
   useEffect(() => {
-    return editor.registerUpdateListener(({ editorState }) => {
-      const json = JSON.stringify(editorState.toJSON());
-      if (updateContent) {
-        updateContent(json);
-      }
+    return editor.registerUpdateListener(() => {
+      editor.update(() => {
+        if (updateContent) {
+          // TODO: Empty posts still have a parent node - can this be detected and avoided?
+          updateContent($generateHtmlFromNodes(editor, null));
```

Comment:
NOTE: For "editing" posts, we'll need to take the generated HTML returned from the API call and transform it back into something the the `@lexical` editor understands: https://lexical.dev/docs/concepts/serialization#lexical---html